addons/web/static/tests/views/fields/reference_field_tests.js

```diff
@@ -964,6 +966,71 @@ QUnit.module("Fields", (hooks) => {
         }
     );
 
+    QUnit.test(
+        "Change model_field value on ReferenceField with model_field option and select invalid value (tree list in form view)",
+        async function (assert) {
+            serverData.models.turtle.records[0].partner_ids = [1];
+            serverData.models.partner.records[0].reference = "product,41";
+            serverData.models.partner.records[0].model_id = 20;
+
+            let updateCount = 0;
+            patchWithCleanup(ReferenceField.prototype, {
+                setup() {
+                    this._super();
+                    onWillRender(() => {
+                        updateCount++;
+                        if (updateCount > 100) {
+                            // we need to throw an error directly for the test to fail
+                            throw Error("ReferenceField has entered an infinite render loop");
+                        }
+                    });
+                }
+            });
+
+            await makeView({
+                type: "form",
+                resModel: "turtle",
+                resId: 1,
+                serverData,
+                arch: `
+                    <form>
+                        <field name="partner_ids">
+                            <tree editable="bottom">
+                                <field name="name" />
+                                <field name="model_id"/>
+                                <field name="reference" required="true" options="{'model_field': 'model_id'}" class="reference_field" />
+                            </tree>
+                        </field>
+                   </form>`,
+            });
+            assert.strictEqual(target.querySelector(".reference_field").textContent, "xpad");
+            assert.strictEqual(target.querySelector(".o_list_many2one").textContent, "Product");
+
+            await click(target, ".o_list_table td.o_list_many2one");
+            await click(target, ".o_list_table .o_list_many2one input");
+            // make sure the patch is working
+            assert.ok(updateCount);
+            // make sure the options are there
+            assert.ok(1 < [...target.querySelectorAll(".o_list_table .o_list_many2one ul li")].length);
+            // Select first option
+            await triggerEvent(target, ".o_list_table .o_list_many2one input", "keydown", {
+                key: "Enter",
+            });
+            await click(target, ".o_list_table .o_list_many2one input");
+            assert.strictEqual(target.querySelector(".reference_field input").value, "");
+            assert.strictEqual(target.querySelector(".o_list_many2one input").value, "Partner");
+            // 'False' value in required reference to make invalid
```

Comment:
Would put a backspace before it and say something along the lines of:

"Now void the associated "reference" field, which is required, and make sure the form marks the field as required"